File: example/old/binarystep.py
import numpy as np

class binarystep:
  def __repr__(self):
    return 'binarystep()'

  def forward(self, input):
    equation = np.vectorize(self.equationForbinaryStep, otypes=[float])
    return equation(input)

  def equationForbinaryStep(self, input):
    return 0 if (input < 0) else 1
    
  def backward(self, input):
    return 0
    # if input == 0 return undefined, this was removed because it would return an error 

File: efficiency-testing/aiinpycpp/binarystep.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class binarystep {
	public:	
		vector <double> forward(vector <double> input) {
      transform(input.begin(), input.end(), input.begin(), [](double input){ return (input < 0) ? 0 : 1; });
      return input;
		}

		vector <double> backward(vector <double> input) {
			return vector <double> (input.size(), 1);
		}
};
